### PR TITLE
fix: Focus handling with a scope on the `GameWidget`

### PR DESCRIPTION
**Pull request by renancaraujo**
# Description

This ignores keyboard events if the game widget doesn't have the primary focus. This allows focus nodes on overlay widgets to take precedence on keyboard resolving.

Also, add focus scope in the game widget for when the overlay is removed, the focus goes back to the game. 

Also, add tests for `GameWidget` for all these cases and some more.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

## Related Issues

This is related to:
- https://github.com/flame-engine/flame/issues/1710
- https://github.com/flame-engine/flame/pull/1719

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame/test/game/game_widget/game_widget_test.dart [hunk 1]
```diff
@@ -221,4 +238,242 @@ void main() {
       expect(game2, isNull);
     });
   });
+  group('focus', () {
```
Comment (spydon) [inline packages/flame/test/game/game_widget/game_widget_test.dart:241]:
```suggestion

  group('focus', () {
```

**Review by spydon — APPROVED** [review]
Nice!!

File: packages/flame/test/game/game_widget/game_widget_test.dart [hunk 2]
```diff
@@ -221,4 +238,242 @@ void main() {
...
+    testWidgets('autofocus starts focused', (tester) async {
+      final gameFocusNode = FocusNode();
+
+      await tester.pumpWidget(
+        GameWidget(
+          focusNode: gameFocusNode,
+          game: FlameGame(),
+          // ignore: avoid_redundant_argument_values
+          autofocus: true,
```
Comment (erickzanardo) [inline packages/flame/test/game/game_widget/game_widget_test.dart:250]:
```suggestion
```

**Review by erickzanardo — APPROVED** [review]
One nit, but LGTM! Can we have a more contextual PR title though?

File: packages/flame/test/game/game_widget/game_widget_test.dart (hunk 2 above)
Comment (spydon) [reply packages/flame/test/game/game_widget/game_widget_test.dart:251]: It's pretty good to keep this imho, it gives the reader of the test more context of what is being tested.

Comment (renancaraujo) [reply packages/flame/test/game/game_widget/game_widget_test.dart:251]: Disagree, that is to explicitly show what that test tests.

**Pull request merged**